### PR TITLE
fix(custom-resources): Fix error handling via custom resources

### PR DESCRIPTION
**Pull request by scott-mccracken**
When a custom resource fails to create/update using the custom resource module, the custom resource is deleted by CloudFormation and further updates are not tracked.

This is caused by the Custom Resource Lambda returning the logStreamName  rather than the physicalResourceId on failure

For further context, please see the following issue https://github.com/aws/aws-cdk/issues/13527

Fixes https://github.com/aws/aws-cdk/issues/13527
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by rix0rrr — CHANGES_REQUESTED** [review]
Thanks for this change, but it also needs a test!

File: packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts [hunk 1]
```diff
@@ -112,7 +112,25 @@ function patchSdk(awsSdk: any): any {
 }
 
 /* eslint-disable @typescript-eslint/no-require-imports, import/no-extraneous-dependencies */
-export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent, context: AWSLambda.Context) {
+export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent, _context: AWSLambda.Context) {
+  console.log(JSON.stringify(event));
+  event.ResourceProperties.Create = decodeCall(event.ResourceProperties.Create);
+  event.ResourceProperties.Update = decodeCall(event.ResourceProperties.Update);
+  event.ResourceProperties.Delete = decodeCall(event.ResourceProperties.Delete);
```
Comment (rix0rrr) [inline packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts:119]: If any of these fail the same error can still happen. The only safe initialization I think is:

```
let physicalResourceId = event.PhysicalResourceId ?? context.logStreamName;
```

And if we do that, all the other code changes aren't necessary anymore, correct?

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts (hunk 1 above)
Comment (scott-mccracken) [reply packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts]: Thanks @rix0rrr - the PhysicalResourceID isn't accessible for create events so generates the following error:
`Property 'physicalResourceId' does not exist on type 'CloudFormationCustomResourceCreateEvent'`

I've updated it to default to context.logStreamName and update as it was before in the pipeline, let me know if there's issues with this approach.

Have we got any existing tests that need updating? Or do we have any examples of how to go about testing the runtime lambda?

Comment (rix0rrr) [reply packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts]: I see. In order to properly compile here, we would need a type test.

You should be able to do this:

```ts
let physicalResourceId = event.RequestType === 'Create' ? context.logStreamName : event.PhysicalResourceId;
```

**Review by rix0rrr — CHANGES_REQUESTED** [review]
Setting to RC again

**Pull request closed without merging**